src/modules/peek/Peek.FilePreviewer/Previewers/ImagePreviewer/ImagePreviewer.cs

```diff
@@ -118,9 +114,11 @@ private Task<bool> LoadLowQualityThumbnailAsync()
                         throw new ArgumentNullException(nameof(hbitmap));
                     }
 
+                    CancellationToken.ThrowIfCancellationRequested();
+
                     await Dispatcher.RunOnUiThread(async () =>
                     {
-                        var thumbnailBitmap = await GetBitmapFromHBitmapAsync(hbitmap);
+                        var thumbnailBitmap = await GetBitmapFromHBitmapAsync(hbitmap, CancellationToken);
```

Comment:
Consider adding a `ThrowIfCancellationRequested` before doing this. There might be a delay before this actually runs on the UI thread as the dispatcher only queues the work.